crates/circuit/src/operations.rs

```diff
@@ -2759,6 +2932,20 @@ impl Operation for UnitaryGate {
     fn directive(&self) -> bool {
         false
     }
+    fn matrix_as_static_1q(&self, _params: &[Param]) -> Option<[[Complex64; 2]; 2]> {
+        match self.num_qubits() {
+            1 => match &self.array {
+                ArrayType::NDArray(arr) => {
+                    Some([[arr[(0, 0)], arr[(0, 1)]], [arr[(1, 0)], arr[(1, 1)]]])
+                }
+                ArrayType::OneQ(mat) => {
+                    Some([[mat[(0, 0)], mat[(0, 1)]], [mat[(1, 0)], mat[(1, 1)]]])
+                }
+                _ => None,
+            },
+            _ => None,
+        }
+    }
```

Comment:
I'd be tempted to invert the logic here so it's:

```suggestion
        match &self.array {
            ArrayType::OneQ(mat) => {
                Some([[mat[(0, 0)], mat[(0, 1)]], [mat[(1, 0)], mat[(1, 1)]]])
            }
            ArrayType::NDArray(arr) => {
                if self.num_qubits() == 1 {
                    Some([[arr[(0, 0)], arr[(0, 1)]], [arr[(1, 0)], arr[(1, 1)]]])
                } else {
                    None
                }
            }                        
            ArrayType::TwoQ(_) => None,                        
        }
    }
```

It feels slightly more compact and we're avoiding inspecting two conditions in the case of `ArrayType::OneQ` although these checks are so fast that performance doesn't matter.

Reply:
I like the suggested way, done in [948c82e](https://github.com/Qiskit/qiskit/pull/14020/commits/948c82e1244e0a25191cc4a3e284b2c702563150)